output/output.go

```diff
@@ -76,31 +78,56 @@ import (
 // 		the tabular and csv formats.
 func Print(c *cli.Context, f *func() interface{}, keys []string) {
 	i := (*f)()
+	keys = limitFields(c, keys)
+	w := c.App.Writer
 	if c.IsSet("json") {
-		jsonOut(i)
+		switch i.(type) {
+		case map[string]interface{}:
+			m := i.(map[string]interface{})
+			metadataJSON(w, m, keys)
+		case []map[string]interface{}:
+			m := i.([]map[string]interface{})
+			listJSON(w, m, keys)
+		default:
+			defaultJSON(w, i)
+		}
 		return
 	}
 	if c.IsSet("csv") {
 		switch i.(type) {
 		case map[string]interface{}:
 			m := i.(map[string]interface{})
-			metadataCSV(c, m, keys)
+			metadataCSV(w, m, keys)
 		case []map[string]interface{}:
 			m := i.([]map[string]interface{})
-			listCSV(c, m, keys)
+			listCSV(w, m, keys)
 		default:
-			fmt.Fprintf(c.App.Writer, "%v", i)
+			fmt.Fprintf(w, "%v", i)
 		}
 		return
 	}
 	switch i.(type) {
 	case map[string]interface{}:
 		m := i.(map[string]interface{})
-		metadataTable(c, m, keys)
+		metadataTable(w, m, keys)
 	case []map[string]interface{}:
 		m := i.([]map[string]interface{})
-		listTable(c, m, keys)
+		listTable(w, m, keys)
 	default:
-		fmt.Fprintf(c.App.Writer, "%v", i)
+		fmt.Fprintf(w, "%v", i)
+	}
+}
+
+func limitFields(c *cli.Context, keys []string) []string {
+	if c.IsSet("fields") {
+		fields := strings.Split(strings.ToLower(c.String("fields")), ",")
```

Comment:
True. In general, no. For example, passing a `--Fields` flag will give an `Incorrect Usage` error and print out the flag options.